openverse_catalog/dags/common/extensions.py

```diff
@@ -0,0 +1,17 @@
+from typing import Literal
+
+
+EXTENSIONS = {
+    "image": {"jpg", "jpeg", "png", "gif", "bmp", "webp", "tiff", "tif", "svg", "webp"},
+    "audio": {"mp3", "ogg", "wav", "aiff", "flac", "aac", "m4a", "wma", "mp4", "m4b"},
```

Comment:
I wonder if treating `m4a` and `mp4` as distinct file types for _audio_ is safe. `mp4` files that are audio only are _supposed_ to be rare, m4a is really the correct extension for those types of files. `mp4` is generally used for audio + video (and maybe subtitles and other things), i.e., an AV container format.

It makes me wonder if our search would be a lot more useful if we didn't use just file extensions but actually merged into _format_, which is more useful information than filetype. It probably requires more work though than just checking out the URL (for example, actually downloading each file and checking the headers manually or with tools). Wikimedia has some good code we can check out in this regard.

It'd probably be more useful, for example, to merge `m4a` and audio-only `mp4`s into a general MPEG-4 Part 14 "format". We'd get even more mileage out of this if we extracted container information like the audio format in ogg files or the type of wma, wma uses the same extension for [wma's mp3 equivalent (lossy), wma's lossless implementation, and two other lesser used wma codecs](https://en.wikipedia.org/wiki/Windows_Media_Audio#Codecs). In a sense the `wma` "filetype" is actually not useful in many cases... maybe you only care to check out lossless files, in that case you'd have to comb through Oggs, WMAs and FLACs, and only one of those is definitely lossless! The others are only _potentially_ lossless, but our current approach won't help anyone find that information out without manually downloading and inspecting each one.

I'm not sure it is something we can solve today, maybe we'd need a lot more work to do this, but I think it is important to consider the meaning behind these filetypes and how we convey them to users and what users could be looking for. If we just extract the "filetype" from the extension, then we probably shouldn't call it filetype and should call it "extension", it's more accurate that way, at least for audio.

Thankfully images (I think) are more straightforward (though there are multiple JPEG standards) and the extension does generally correspond 1-to-1 to a specific format.

Reply:
Thank you for all the context about the `filetype`/`format`/`extension` meanings, @sarayourfriend!
Extracting the `format` information can be a project for some time in the future as it would require a lot of changes in how we download the data.
I think a good interim solution could be to change the frontend wording for audio to use `extension` instead of `filetype`, and add a note about extensions not always corresponding to the format of the API.